processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -584,7 +584,7 @@ IncrementalIndexRowResult toIncrementalIndexRow(InputRow row)
           wasNewDim = true;
           final DimensionHandler<?, ?, ?> handler;
           if (useSchemaDiscovery) {
-            handler = new NestedCommonFormatColumnHandler(dimension);
+            handler = new NestedCommonFormatColumnHandler(dimension, null);
```

Comment:
do we not need to pass the desired type here?

Reply:
there is no desired type here since the dimension was discovered on the fly

Reply:
oh ok. this is the `wasNewDim = true` case.